### ISSUE TITLE
Allow updating dependencies in "conandata.yml"

### ISSUE THREAD

**Issue by gsemet** [renovatebot/renovate#21300]
### What would you like Renovate to be able to do?

I have deported the declaration of dependencies of my conan packages into the conandata.yml that is automatically supported by conan.

It looks like this:

{code}
requirements:
  - my-lib/1.2.3@user/channel
  - my-other-lib/[>1.0]@user/channel
tools_requirements:
  - my-tool/1.2.3@user/channel
  - my-other-tool/[>1.0]@user/channel
 {code}

### If you have any ideas on how this should be implemented, please tell us here.

Provide an option to parse the yaml file. It should be pretty similar than patching the conanfile.txt for instance.

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
Need to decide if it's a new package manager or reuse existing 

Where is it documented?

**Comment by HonkingGoose** [comment]
> Where is it documented?

I think I found some relevant information about the `conandata.yml` file. The docs have different pages for each version of Conan, so check what version of the docs you're reading! 😉 

Pages for the `2.x` version of Conan:

- https://docs.conan.io/2/reference/conanfile/attributes.html#conan-data
- https://docs.conan.io/2/tutorial/creating_packages/handle_sources_in_packages.html#using-the-conandata-yml-file

Page for the `1.59` version of Conan:

- https://docs.conan.io/en/1.59/reference/config_files/conandata.yml.html

**Comment by gsemet** [comment]
Yes, this is that. The thing is that conandata.yml is automatically exported when it exists, so it makes it easier to integrate into a recipe. But you still need to read it and execute the self.requires manually in the recipe, sadly this is not supported by default by conan.

But this makes sooo much easier and safer for external parsers to patch it. We use it in the company i am working on to automatically update the requirements and tools requirements, but if renovate can do it, we can ditch our code basically.

**Comment by rarkins** [comment]
I don't really understand what we should do with the contents after we've parsed the file.

e.g.

```yml
sources:
  "1.0":
    url: "https://github.com/conan-io/libhello/archive/refs/heads/main.zip"
    sha256: "7bc71c682895758a996ccf33b70b91611f51252832b01ef3b4675371510ee466"
    strip_root: true
```

What is the dependency name here? It seems to be a fake release for the example.. yet it's tagged as 0.0.1 and not 1.0: https://github.com/conan-io/libhello/releases
